jetbrains-core/src/software/aws/toolkits/jetbrains/services/telemetry/ClientTelemetryService.kt

```diff
@@ -27,23 +28,55 @@ import java.net.URI
 import java.time.Duration
 import java.time.Instant
 
-interface ClientTelemetryService : Disposable
+interface ClientTelemetryService : Disposable {
+    fun recordMetric(metricNamespace: String): Metric
+}
 
-class DefaultClientTelemetryService(sdkClient: AwsSdkClient, regionProvider: ToolkitRegionProvider) : ClientTelemetryService {
+sealed class DefaultClientTelemetryService(
+    private val sdkClient: AwsSdkClient,
+    private val regionProvider: ToolkitRegionProvider
+) : ClientTelemetryService {
     private lateinit var startupTime: Instant
```

Comment:
No reason for this to be late inint, you are creating it in the `init{}` anyway. assign it in init, then reference it in the `use` block